### ISSUE TITLE
RDS database resource stays in Synced: False state, error: No modifications were requested

### ISSUE THREAD

**Issue by bandesz** [crossplane-contrib/provider-aws#480]
### What happened?

When creating an RDSInstance resource, it stays in Synced: false state and has the following error:

```
  conditions:
  - lastTransitionTime: "2021-01-04T13:42:52Z"
    message: "update failed: cannot modify RDS instance: InvalidParameterCombination:
      No modifications were requested\n\tstatus code: 400, request id: c45a65b8-7215-4115-b8ac-f38b2332bcea"
    reason: ReconcileError
    status: "False"
    type: Synced
```

The relevant event says exactly the same error.

```
116s        Warning   CannotUpdateExternalResource   rdsinstance/example-postgres-test     (combined from similar events): cannot modify RDS instance: InvalidParameterCombination: No modifications were requested
            status code: 400, request id: e437cc95-e01d-4805-bfcf-96450565b853
```

The error doesn't affect the claim which has Ready: True.

### How can we reproduce it?

Composition used:

```
apiVersion: apiextensions.crossplane.io/v1
kind: Composition
metadata:
  annotations:
    helm.fluxcd.io/antecedent: example:helmrelease/eks-dev-crossplane-aws-rds
  creationTimestamp: "2021-01-04T12:19:06Z"
  generation: 1
  labels:
    crossplane-configuration: crossplane-aws-rds
    provider: aws
  name: aws-rds-postgresql-database-default
  resourceVersion: "17750"
  selfLink: /apis/apiextensions.crossplane.io/v1/compositions/aws-rds-postgresql-database-default
  uid: a373b85a-3de7-4af1-8a9e-321bbf9877a9
spec:
  compositeTypeRef:
    apiVersion: aws.crossplane.example.com/v1alpha1
    kind: PostgreSQLDatabase
  resources:
  - base:
      apiVersion: database.aws.crossplane.io/v1beta1
      kind: DBSubnetGroup
      spec:
        forProvider:
          description: null
          region: eu-west-2
          subnetIds:
          - subnet-1
          - subnet-2
          - subnet-3
    patches:
    - fromFieldPath: spec.claimRef.name
      toFieldPath: metadata.name
      transforms:
      - string:
          fmt: example-%s
        type: string
      type: FromCompositeFieldPath
    - fromFieldPath: spec.claimRef.name
      toFieldPath: spec.forProvider.description
      transforms:
      - string:
          fmt: DB Subnet Group for example-%s
        type: string
      type: FromCompositeFieldPath
    - fromFieldPath: spec.parameters.deletionPolicy
      toFieldPath: spec.deletionPolicy
      type: FromCompositeFieldPath
  - base:
      apiVersion: ec2.aws.crossplane.io/v1beta1
      kind: SecurityGroup
      spec:
        forProvider:
          deletionPolicy: Delete
          description: Allow access to PostgreSQL from within the VPC
          ingress:
          - fromPort: 5432
            ipProtocol: tcp
            ipRanges:
            - cidrIp: 10.0.0.0/16
            toPort: 5432
          region: eu-west-2
          vpcId: vpc-xxx
    patches:
    - fromFieldPath: metadata.uid
      toFieldPath: metadata.name
      type: FromCompositeFieldPath
    - fromFieldPath: metadata.uid
      toFieldPath: spec.forProvider.groupName
      transforms:
      - string:
          fmt: example-%s-vpc
        type: string
      type: FromCompositeFieldPath
    - fromFieldPath: spec.parameters.deletionPolicy
      toFieldPath: spec.deletionPolicy
      type: FromCompositeFieldPath
  - base:
      apiVersion: database.aws.crossplane.io/v1beta1
      kind: RDSInstance
      spec:
        forProvider:
          copyTagsToSnapshot: true
          dbSubnetGroupNameSelector:
            matchControllerRef: true
          engine: postgres
          masterUsername: xxx
          publiclyAccessible: false
          region: eu-west-2
          vpcSecurityGroupIDSelector:
            matchControllerRef: true
        writeConnectionSecretToRef:
          namespace: example
    connectionDetails:
    - fromConnectionSecretKey: username
    - fromConnectionSecretKey: password
    - fromConnectionSecretKey: endpoint
    - fromConnectionSecretKey: port
    patches:
    - fromFieldPath: spec.claimRef.name
      toFieldPath: metadata.name
      transforms:
      - string:
          fmt: example-%s
        type: string
      type: FromCompositeFieldPath
    - fromFieldPath: metadata.uid
      toFieldPath: spec.writeConnectionSecretToRef.name
      transforms:
      - string:
          fmt: '%s-rdsinstance'
        type: string
      type: FromCompositeFieldPath
    - fromFieldPath: spec.parameters.allocatedStorage
      toFieldPath: spec.forProvider.allocatedStorage
      type: FromCompositeFieldPath
    - fromFieldPath: spec.parameters.allowMajorVersionUpgrade
      toFieldPath: spec.forProvider.allowMajorVersionUpgrade
      type: FromCompositeFieldPath
    - fromFieldPath: spec.parameters.applyModificationsImmediately
      toFieldPath: spec.forProvider.applyModificationsImmediately
      type: FromCompositeFieldPath
    - fromFieldPath: spec.parameters.autoMinorVersionUpgrade
      toFieldPath: spec.forProvider.autoMinorVersionUpgrade
      type: FromCompositeFieldPath
    - fromFieldPath: spec.parameters.backupRetentionPeriod
      toFieldPath: spec.forProvider.backupRetentionPeriod
      type: FromCompositeFieldPath
    - fromFieldPath: spec.parameters.characterSetName
      toFieldPath: spec.forProvider.characterSetName
      type: FromCompositeFieldPath
    - fromFieldPath: spec.parameters.dbInstanceClass
      toFieldPath: spec.forProvider.dbInstanceClass
      type: FromCompositeFieldPath
    - fromFieldPath: spec.parameters.dbName
      toFieldPath: spec.forProvider.dbName
      type: FromCompositeFieldPath
    - fromFieldPath: spec.parameters.deletionPolicy
      toFieldPath: spec.deletionPolicy
      type: FromCompositeFieldPath
    - fromFieldPath: spec.parameters.deletionProtection
      toFieldPath: spec.forProvider.deletionProtection
      type: FromCompositeFieldPath
    - fromFieldPath: spec.parameters.engineVersion
      toFieldPath: spec.forProvider.engineVersion
      type: FromCompositeFieldPath
    - fromFieldPath: spec.claimRef.name
      toFieldPath: spec.forProvider.finalDBSnapshotIdentifier
      transforms:
      - string:
          fmt: example-%s-final-snapshot
        type: string
      type: FromCompositeFieldPath
    - fromFieldPath: spec.parameters.iops
      toFieldPath: spec.forProvider.iops
      type: FromCompositeFieldPath
    - fromFieldPath: spec.parameters.multiAZ
      toFieldPath: spec.forProvider.multiAZ
      type: FromCompositeFieldPath
    - fromFieldPath: spec.parameters.preferredBackupWindow
      toFieldPath: spec.forProvider.preferredBackupWindow
      type: FromCompositeFieldPath
    - fromFieldPath: spec.parameters.preferredMaintenanceWindow
      toFieldPath: spec.forProvider.preferredMaintenanceWindow
      type: FromCompositeFieldPath
    - fromFieldPath: spec.parameters.skipFinalSnapshotBeforeDeletion
      toFieldPath: spec.forProvider.skipFinalSnapshotBeforeDeletion
      type: FromCompositeFieldPath
    - fromFieldPath: spec.parameters.storageType
      toFieldPath: spec.forProvider.storageType
      type: FromCompositeFieldPath
  writeConnectionSecretsToNamespace: example
```

### What environment did it happen in?

Crossplane version: 1.0.0
AWS provider: 0.16.0

**Comment by danports** [comment]
I just ran into this issue with a similar composition. I configured CloudTrail to log API calls and noticed that Crossplane was constantly calling `ModifyDBInstance` with these request parameters:
```json
"dBInstanceIdentifier": "my-database-id",
"applyImmediately": false,
"preferredMaintenanceWindow": "Sat:08:00-Sat:09:00",
"allowMajorVersionUpgrade": false
```
The `preferredMaintenanceWindow` property was there because the AWS API always returns the value in lowercase (e.g. `sat:08:00-sat:09:00`) but I had specified the value in title case in my resource. When I changed the case in my resource to eliminate that spurious change, I started receiving the `No modifications were requested` error. I assume (but haven't yet verified) that the AWS provider is now calling `ModifyDBInstance` with the following parameters, and the AWS API doesn't like it:
```json
"dBInstanceIdentifier": "my-database-id",
"applyImmediately": false,
"allowMajorVersionUpgrade": false
```
